### PR TITLE
If we're knocked out, we'll drop any object that we're holding 

### PR DESCRIPTION
**Pull request by vinnytherabbit**
If we're knocked out and we're holding a object, we'll drop it immediately

### CODE REVIEW

**Comment by Dliwk** [comment]
That does look like something that might work (:tada:), but right now that's not true.

File: src/ballistica/scene/node/spaz_node.cc [hunk 1]
```diff
@@ -3884,6 +3884,11 @@ void SpazNode::Step() {
     if (!holding_something_ && hold_node_.exists()) hold_node_.Clear();
   }
 
+  // If we're knocked out, stop any mid-progress pick up.
+  if (knockout_) {
+    holding_something_ = false;
```
Comment (Dliwk) [inline src/ballistica/scene/node/spaz_node.cc:3889]: I've just got this working by replacing this line with 
```c++
DropHeldObject();
```
Think about whether it suits you.

**Review by Dliwk — APPROVED** [review]

**Comment by IlScottlI** [comment]
Cannot wait for this update!

**Comment by efroemling** [comment]
Thanks for the contribution!

This does look like a reasonable change. The only pitfall is that it affects the C++ level simulation, which gets run by all clients in a networked game - meaning that if some clients are running older versions of the game and some are running a newer version with this change, they'll see different behavior which will lead to ugly physics corrections and snapping for some people.

There are ways to deal with this but they're not especially simple (see the behavior_version attribute on the spaz node).

One of my big goals for the next update (1.8) is scene-versions which will let us split out all the c++ level sim stuff into different versions where we can make logic changes like this without worrying about breaking older versions, so I vote we sit on this change for now and try to get it in for 1.8.

**Pull request closed without merging**